feat: how to enable host network. (#731)
kubewarden/docs: docs/howtos/host-network.md

## What changes when host network is enabled

When `hostNetwork` is enabled, all Kubewarden pods share the host node's
network namespace instead of having their own isolated network. This has some
implications.

**Ports are bound on the host.** Each port used by a Kubewarden component
occupies a port on the node's network interface, not in a pod-private
namespace. Two pods that try to bind the same port on the same node will
conflict.

**Increased attack surface.** Processes inside host-network pods can see all
network interfaces of the host node, and webhook endpoints become reachable
outside the cluster. See the [Security considerations](#security-considerations)
section below.

## Custom port configuration

To help avoid port conflicts in host network mode, the `PolicyServer` CRD
provides fields for overriding the default ports used by each PolicyServer
instance:

- **`spec.webhookPort`** — The port used by the PolicyServer to serve webhook
  HTTPS requests. Defaults to `8443`.
- **`spec.readinessProbePort`** — The port used for the readiness probe HTTP
  endpoint. Defaults to `8081`.
- **`spec.metricsPort`** — The port used to expose metrics. Defaults to `8080`.

> These fields are useful in host network mode for two main reasons:

Reviewer: ```suggestion
These fields are useful in host network mode for two reasons:
```

1. **Running multiple PolicyServers on the same node.** By assigning different
   ports to each PolicyServer, you can schedule them on the same node without
   port conflicts.
2. **Avoiding conflicts with other host services.** If another service on the
   node already uses a default port, you can change the PolicyServer port to
   avoid the collision.

### Example: PolicyServer with custom ports

```yaml
apiVersion: policies.kubewarden.io/v1
kind: PolicyServer
metadata:
  name: custom-ports
spec:
  image: ghcr.io/kubewarden/policy-server:latest
  replicas: 2
  webhookPort: 9443
  readinessProbePort: 9081
  metricsPort: 9080
```

## Affinity configuration

> When running in host network mode, if two Kubewarden pods land on the same node
> and try to bind the same port, one of them will fail to start and enter a crash
> loop. That's where the anti-affinity configuration came to the rescue. Without
> anti-affinity rules:

Reviewer: ```suggestion
When running in host network mode, if two Kubewarden pods land on the same node
and try to bind the same port, one of them will fail to start and enter a crash
loop. This is where the anti-affinity configuration comes to the rescue. Without
anti-affinity rules:
```

---
sidebar_label: Host network mode
title: Running Kubewarden in host network mode
description: How to enable and configure host network mode for the Kubewarden stack.

> sidebar_position: 23

Reviewer 2: I would put it below around raw policies, and not as high as it is right now (between "audit scanner" and "configuring policies").

```suggestion
sidebar_position: 71
```
